packages/cli/src/pages/api/sendMail.ts

```diff
@@ -1,111 +1,64 @@
-import React from "react";
 import type { NextApiRequest, NextApiResponse } from "next";
-import { render } from "mailing-core";
 import { MjmlError } from "mjml-react";
-import { templates, sendMail } from "../../moduleManifest";
+import { sendMail } from "../../moduleManifest";
+import renderTemplate from "../../util/renderTemplate";
+import { validateApiKey } from "src/util/validateApiKey";
 
 type Data = {
   error?: string; // api error messages
-  html?: string;
   mjmlErrors?: MjmlError[];
+  result?: any;
 };
 
-function renderTemplate(
-  templateName: string,
-  props: { [key: string]: any }
-): { error?: string; mjmlErrors?: MjmlError[]; html?: string } {
-  const Template = templates[templateName as keyof typeof templates];
-  if (!Template) {
-    return {
-      error: `Template ${templateName} not found in list of templates: ${Object.keys(
-        templates
-      ).join(", ")}`,
-    };
-  }
-
-  return render(React.createElement(Template as any, props as any));
-}
-
-async function validApiKey(apiKey: string | string[] | undefined) {
-  if (!apiKey) return false;
-
-  const host = process.env.MM_DEV ? "http://localhost:3883" : "yourInstallUrl";
-  const response = await fetch(`${host}/api/apiKeys/${apiKey}/validate`);
-
-  return 200 === response.status;
```

Comment:
@alexfarrill I forget... did we want this to be a network call to get potential caching benefits? I switched to hitting the DB directly but then thought maybe that's why we did it. I _think_ just hitting the DB should be OK because we're already going to be hitting it in anything that requires an API key.

Reply:
^ discussed w alex. hitting db directly seems good. to optimize later we can cache/hardcode api keys at build time for instant lookup.